packages/adapter-auto/index.js

```diff
@@ -10,31 +13,55 @@ for (const candidate of adapters) {
 
 		try {
 			module = await import(candidate.module);
-
-			fn = () => {
-				const adapter = module.default();
-				return {
-					...adapter,
-					adapt: (builder) => {
-						builder.log.info(`Detected environment: ${candidate.name}. Using ${candidate.module}`);
-						return adapter.adapt(builder);
-					}
-				};
-			};
-
-			break;
 		} catch (error) {
 			if (
 				error.code === 'ERR_MODULE_NOT_FOUND' &&
 				error.message.startsWith(`Cannot find package '${candidate.module}'`)
 			) {
-				throw new Error(
-					`It looks like ${candidate.module} is not installed. Please install it and try building your project again.`
-				);
+				try {
+					console.log(`Installing ${candidate.module} on the fly...`);
+					execSync(
+						`${process.platform === 'win32' ? 'set' : ''} NODE_ENV=ignore_me && npm install ${
+							candidate.module
+						} --no-save --omit=dev --no-package-lock`,
+						{
+							stdio: 'inherit',
+							cwd: dirname(fileURLToPath(import.meta.url))
+						}
+					);
```

Comment:
We should be able to manually pass env vars with the `env` option https://nodejs.org/dist/latest-v19.x/docs/api/child_process.html#child_processexecsynccommand-options

If we do that, we can also avoid involving the shell at all, and we can use `execFileSync` here instead.

Reply:
`execFileSync` needs `.cmd` endings for the `npm/pnpm/yarn` command on windows last time I tested this. So we either have to prepend that on windows or keep using `execSync`

Reply:
Yeah I saw your comment [in that thread](https://github.com/sveltejs/kit/pull/7462#discussion_r1011831808) after I commented here. I responded over on that thread. Even if we can't make `execFileSync` work, it would be nice to avoid the manual setting of the env var using shell commands.

I'm actually not even sure that `FOO=bar && do-something` would call `do-something` with `FOO` set to `bar`. I don't think environment variables automatically inherit that way in shell scripts without an `export`.